repos/spack_repo/builtin/packages/qt/package.py

```diff
@@ -575,15 +579,47 @@ def patch(self):
             with open(conf_file, "a") as f:
                 f.write("QMAKE_CXXFLAGS += -std=gnu++98\n")
 
+    def _quoted(self, args):
+        """Returns args with each arg in double quotes if neccesary
+        Necessity determined by:
+            - path with a space on any platform
+            - path with a reserved character on Windows
+        """
+
+        def quote(arg):
+            return '"' + arg + '"'
+
+        def has_space(arg):
+            return " " in arg
+
+        def has_reserved(arg):
+            if not IS_WINDOWS:
+                return False
+            return True if re.search(r"[ <>^:\"|?*]", arg) else False
+
+        return [quote(x) if has_space(x) or has_reserved(x) else x for x in args]
+
+    def _split_link_args(self, arg_str):
+        """Returns a list of the -L
+        arguments included in arg_str with proper
+        handling for paths with spaces"""
+        return ["-L" + x for x in arg_str.split("-L") if x]
```

Comment:
I think this assumes no path has `-L` "in the middle" of it.

`search_flags`/`include_flags` are `-L`/`-I` of `.directories` (of the respective objects), so it would be safer to access those directly IMO

Reply:
Hmm, that a good point that `my-Library` is absolute a legal path that could be in an install dir. 
I'll refactor. See also https://github.com/spack/spack/pull/51078 where I try to resolve this for the general case, not just Qt, without resorting to this